extensions/items/standalone.mjs

```diff
@@ -0,0 +1,333 @@
+   * 2. Retrieve `info.json` (which contains the basic information of an item).
+   * 3. Create an item instance (which contains all possible functions that can be performed on the item)
+   */
+  async initialize() {
+    const allItemsName = await fs.promises.readdir('../extensions/items/common/itemClasses');
+    const itemSettingJson = await fs.promises.readFile(`../extensions/items/common/config.json`);
+    const itemConfig = JSON.parse(itemSettingJson).items;
+    const itemBaseClasses = {};
+    for (let itemTypeName of allItemsName) {
+      const itemBaseModule = await import(`../items/common/itemClasses/${itemTypeName}/${itemTypeName}.mjs`);
+      const itemBaseClass = itemBaseModule.default;
+      itemBaseClasses[itemTypeName] = itemBaseClass;
+    }
+    let index = 0;
+    for (let itemName in itemConfig) {
+      const itemBaseClassName = itemConfig[itemName].baseClass;
+      const item = new itemBaseClasses[itemBaseClassName](this.helper, itemConfig[itemName].imagePath);
+      this.itemInfo[itemName] = {
+        index: index,
+        layer: itemConfig[itemName].layer,
+        show: item.show,
+        exchangeable: item.exchangeable,
+        usable: item.usable
+      }
+      this.itemInstances[itemName] = item;
+      index++;
+
+      /* Initialize cell set for each items */
+      for (const mapName of this.helper.gameMap.getOriginalCellSetStartWith('').keys()) { // Get all available maps.
+        const cellSet = {
+          mapName: mapName,
+          name: 'droppedItem' + itemName.toUpperCase(),
+          priority: 4,
+          cells: Array.from(this.droppedItemCell.values()),
+          layers: [
+            { "item": this.itemInfo[itemName].layer }
+          ],
+        };
+        this.helper.gameMap.setDynamicCellSet(mapName, cellSet);
+      }
+    }
+  }
+
+  /**
+   * Client initialization. This function will set the basic information of a player's item to an empty object
+   * The below function should be called whenever a player starts a game session.
+   */
+  async c2s_playerInit(player) {
+    if (!this.items.has(playerID)) {
+      this.items.set(playerID, {});
+    }
+  }
+
+  /**
+   * This function will return an object that contains the basic information of all items.
+   * @return {object} partials - An object with the following type:
+   * type ItemInfo = {
+   *   [itemName: string]: {
+   *     url: string;
+   *     show: boolean;
+   *     exchangeable: boolean;
+   *     usable: boolean;
+   *     properties: {};
+   *   }
+   * }
+   */
+  async c2s_getItemInfo(playerID) {
+    return this.itemInfo;
+  }
+
+  /**
+   * Returning all items owned by the user
+   * @return {object} partials - An object with the following type:
+   * type ItemObj = {
+   *   [itemName: string]: {
+   *     amount: number;
+   *   }
+   * }
+   */
+  async c2s_getAllItems(playerID) {
+    /* player has not registered yet */
+    if (!this.items.has(playerID)) {
```

Comment:
Is it possible to do auto-registration instead of failing when it's not registered?